examples/vae/input_pipeline.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Input pipeline for VAE dataset."""
+
+import jax
+import jax.numpy as jnp
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+
+class DSBuilder:
+  """Caches DSBuider but only when called."""
+
+  def __init__(self):
+    self._ds_builder = None
+
+  def __call__(self):
+    return self.ds_builder
+
+  @property
+  def ds_builder(self):
+    if not self._ds_builder:
+      _ds_builder = tfds.builder('binarized_mnist')
+      _ds_builder.download_and_prepare()
+      self._ds_builder = _ds_builder
+
+    return self._ds_builder
+
+DS_BUILDER = DSBuilder()
+
+
+def build_train_set(flags) -> iter:
```

Comment:
Let's pass in `batch_size` instead of `flags` here.

We usually like to have specific arguments, rather than passing around namespaces.